src/covjson_pydantic/domain.py

```diff
@@ -62,31 +52,25 @@ class DomainType(str, Enum):
     multi_point = "MultiPoint"
 
 
-class Axes(BaseModel):
-    x: Optional[Union[ValuesAxis[float], CompactAxis]]
-    y: Optional[Union[ValuesAxis[float], CompactAxis]]
-    z: Optional[Union[ValuesAxis[float], CompactAxis]]
-    t: Optional[ValuesAxis[datetime]]
-    composite: Optional[ValuesAxis[Tuple]]
-
-    @root_validator(skip_on_failure=True)
-    def at_least_one_axes(cls, values):
-        if (
-            values["x"] is None
-            and values["y"] is None
-            and values["z"] is None
-            and values["t"] is None
-            and values["composite"] is None
-        ):
+class Axes(CovJsonBaseModel):
+    x: Optional[Union[ValuesAxis[float], CompactAxis]] = None
+    y: Optional[Union[ValuesAxis[float], CompactAxis]] = None
+    z: Optional[Union[ValuesAxis[float], CompactAxis]] = None
+    t: Optional[ValuesAxis[AwareDatetime]] = None
+    composite: Optional[ValuesAxis[Tuple]] = None
+
+    @model_validator(mode="after")
+    def at_least_one_axes(self):
+        if self.x is None and self.y is None and self.z is None and self.t is None and self.composite is None:
             raise ValueError("At least one axis of x,y,z,t or composite must be given.")
-        return values
+        return self
 
 
 class Domain(CovJsonBaseModel):
```

Comment:
This used to have an extra=allow, because it used CovJsonBaseModel. This is also in the spec. At least that is how I interpret [this](https://opengeospatial.github.io/ogcna-auto-review/21-069.html#_070a1d4d-ff49-426b-8f6a-0c40fd059408)

Reply:
Woops.. missed that one in refactor